### ISSUE TITLE
Practical Guide for Beginner

### ISSUE THREAD

**Issue by hzhz2020** [DataCanvasIO/DeepTables#95]
Thank you for providing this excellent toolkit!

I recently started using it to model some tabular data and have found it to be highly flexible and convenient, supporting various models and use cases. **However, as a new user to this toolkit, I’m feeling a bit overwhelmed and unsure of how to proceed.**

I’m working with a regression task on tabular data that has around 100 features, both numerical and categorical, with many missing values.

1. Could you advise on **which models in DeepTable I should start with**? **What are specific hyperparameters of that model I should focus on tuning?**  Would it be better to begin with the preset models, such as DeepFM, rather than individual components like ‘dnn_nets’ as shown in the Regression demo?

2. In the ModelConfig documentation (https://deeptables.readthedocs.io/en/latest/model_config.html), you mention that the value of nets can be “**ANY** combination of component names, preset models, and custom functions.” For instance, in conf = ModelConfig(nets=['linear','dnn_nets','cin_nets','cross_nets']). **what is each component(like ‘cin_nets’, ‘cross_nets’, etc.), is there more concrete information on how they are defined?**

Additionally, can all the components really be combined together, and if so, how does that work? Does the order of the list in nets=['linear','dnn_nets','cin_nets','cross_nets'] indicate the stacking order of these components?

Thank you for your help!

**Comment by oaksharks** [comment]
Hi @hzhz2020 ,

The choices of model are closely related to the data itself. If it is only a simple linear relationship, then DNN can be well fitted. The more information contained in the data, the more complex the model should be used. The model is too simple, the data cannot be well fitted, and the model is too complex and easy to overfit. For most purpose, I recommend using these networks or combining them: [AFM/MultiheadAttention/FGCNN](https://deeptables.readthedocs.io/en/latest/layers.html).  **HPO is also provided**, see [automl for DT](https://deeptables.readthedocs.io/en/latest/automl.html) .

A **component** means a `tensorflow.keras.layers.Layer` , which accepts high-dimensional features and outputs logits, used to produce predictions based on different task types.

I think plotting the network topology generated by DT will help you understand how itworks (including how multiple subnetworks are combined) , see https://github.com/DataCanvasIO/DeepTables/blob/master/deeptables/examples/quickstart.ipynb

**Comment by hzhz2020** [comment]
Thanks! 

Is there a way to set learning rate?

**Comment by oaksharks** [comment]
Create a optimizer, e.g.  : 
```
optimizer = tf.keras.optimizers.Adam(learning_rate=1e-3)
ModelConfig(optimizer=optimizer, ...)
```

**Comment by hzhz2020** [comment]
thanks
